pkgs/stdenv/generic/setup.sh

```diff
@@ -182,31 +182,31 @@ _addRpathPrefix() {
 
 # Return success if the specified file is an ELF object.
 isELF() {
-    local fn=$1
+    local fn="$1"
     local fd
     local magic
     exec {fd}< "$fn"
-    read -r -n 4 -u $fd magic
+    read -r -n 4 -u "$fd" magic
     exec {fd}<&-
-    if [[ $magic =~ ELF ]]; then return 0; else return 1; fi
+    if [[ "$magic" =~ ELF ]]; then return 0; else return 1; fi
 }
 
 # Return success if the specified file is a script (i.e. starts with
 # "#!").
 isScript() {
-    local fn=$1
+    local fn="$1"
     local fd
     local magic
     if ! [ -x /bin/sh ]; then return 0; fi
     exec {fd}< "$fn"
-    read -r -n 2 -u $fd magic
+    read -r -n 2 -u "$fd" magic
```

Comment:
Even before my general PR, my basic is philosophy with bash is

- `"$asdf"` is a string substitution
- `"${asdf[@]}"` is an array string substitution
- `$asdf` is only for intentional re-splitting, and must be paired with a comment

In other words, don't forgo quoting because of a lack of deliminator in the definition. Even if that is obviously true it is still tempting fate, and per my bullets above is misleading because there is no intention of a word-splitting.